### PR TITLE
Functions for r package

### PR DESCRIPTION
**Pull request by Bertlich**
Hi @reviewers, 

this week, I created my shiny app (app.R) and tried to launch it on shinyapps.io. Theoretically, my app should be online available in the cloud. This is the link to the app: https://bertlich.shinyapps.io/weather_app/. Unfortunately, the display of the images and icons does not work. I don't know if that's because of how I store my images (interactively in a www/ folder). The image display works offline, it's just that it doesn't work when I host the images on the server. If you have any idea how to fix that, that would be awesome! 
UPDATE: I read [here ](https://shiny.rstudio.com/articles/persistent-data-storage.html) that it might be due to the fact that I can't save data locally on the shiny server. Do you have any ideas how to fix that? 

Also, I updated little things in my package code (weather_app_functions.R), such as creating a new small function (get_icon) and including some additional stop() commands in case something goes wrong. 

Also: Is it better to commit less often, so that this pull request does not look as messy? Haha.

I am looking forward to your feedback!

### CODE REVIEW

**Comment by fdabl** [comment]
@MaxMaier42 and @SvdStappen please review this pull request!

**Review by MaxMaier42 — APPROVED** [review]
This looks very nice! Good work on getting so much code done in a week! I played around with the app and apart from the picture problem it seems to work nice as well. Unfortunatly, I cannot really give any recommendations on how to add the pictures, I have not done Shiny apps with other things than R plots before.

File: app.R [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+            br(),
+            
+            # display current general weather forecast
+            fluidRow(
+                column(width = 6, 
+                       offset = 3,
+                       verbatimTextOutput("current_weather"),
+                       align = "center")
+            ),
+
```
Comment (SvdStappen) [inline app.R:113]: You could make your code less redundant by adding these four fluidRow()s into one apply function. (Something like apply(c(“current_temp”, “sunrise”), function(i) { fluidRow() with all the output settings here}))

**Review by SvdStappen — COMMENTED** [review]
Hi!
Seems like your app works well in general! I see the image issue you say, and I am not entirely sure if this will work, but maybe you can retrieve the image you want to display directly form the internet? That way, you don’t have to store it somewhere, so you won’t have the local storage issue. It’s a bit like this StackOverflow post: https://stackoverflow.com/questions/53147225/display-images-from-web-in-shiny-r. I hope that solves your issue! 
 Besides that, to make your code a bit more efficient, you could choose to write a function for your output/input, as they are all similar. I commented in the text where this was present. Disadvantage of this is that it might make your code look more complex.. I would understand if you don’t adopt this for that reason. 
Hope it helps!
Best,
Sjanne

**Pull request merged**